sql/core/src/test/scala/org/apache/spark/sql/SubquerySuite.scala

```diff
@@ -1646,4 +1646,25 @@ class SubquerySuite extends QueryTest with SharedSparkSession with AdaptiveSpark
     checkAnswer(df, df2)
     checkAnswer(df, Nil)
   }
+
+  test("Subquery reuse across the whole plan") {
+    val df = sql(
```

Comment:
Can you put the physical plan in a comment? It'll help to see what gets reused.

Reply:
Sure, I added the plan in commit: https://github.com/apache/spark/pull/28885/commits/2b3cde22f62374c25b5748ff2253e168ec4e5d95
Let me know if a sample plan would be beneficial in other test cases too.